deepspeech.cpp

```diff
@@ -69,89 +69,99 @@ extern "C" {
 
     ModelWrapper* New(const char* aModelPath, int* errorOut)
     {
-        return new ModelWrapper(aModelPath, errorOut);
+        auto mw = new ModelWrapper(aModelPath, errorOut);
+        if (*errorOut != DS_ERR_OK) {
+            delete mw;
```

Comment:
Another approach would be to never delete these wrappers in the .cpp file, and instead expose `FreeModelWrapper`, `FreeStreamWrapper`, etc. functions that Go can call when needed. I mostly wanted to avoid having a mix of the two, where sometimes the wrappers would be freed automatically and other times they'd need to be freed explicitly from Go.

Reply:
This is perfect.